apps/emqx_bridge_rabbitmq/src/emqx_bridge_rabbitmq_connector.erl

```diff
@@ -0,0 +1,557 @@
+                }
+            )},
+        {password,
+            hoconsc:mk(
+                typerefl:binary(),
+                #{
+                    default => <<"guest">>,
+                    required => true,
+                    desc => ?DESC("password")
+                }
+            )},
+        {pool_size,
+            hoconsc:mk(
+                typerefl:pos_integer(),
+                #{
+                    default => 8,
+                    required => true,
+                    desc => ?DESC("pool_size")
+                }
+            )},
+        {timeout,
+            hoconsc:mk(
+                emqx_schema:duration_ms(),
+                #{
+                    default => <<"5s">>,
+                    required => true,
+                    desc => ?DESC("timeout")
+                }
+            )},
+        {wait_for_publish_confirmations,
+            hoconsc:mk(
+                boolean(),
+                #{
+                    required => true,
+                    default => true,
+                    desc => ?DESC("wait_for_publish_confirmations")
+                }
+            )},
+        {publish_confirmation_timeout,
+            hoconsc:mk(
+                emqx_schema:duration_ms(),
+                #{
+                    default => <<"30s">>,
+                    required => true,
+                    desc => ?DESC("timeout")
+                }
+            )},
+
+        {virtual_host,
+            hoconsc:mk(
+                typerefl:binary(),
+                #{
+                    default => <<"/">>,
+                    required => true,
+                    desc => ?DESC("virtual_host")
+                }
+            )},
+        {heartbeat,
+            hoconsc:mk(
+                emqx_schema:duration_ms(),
+                #{
+                    default => <<"30s">>,
+                    required => true,
+                    desc => ?DESC("heartbeat")
+                }
+            )},
+        {auto_reconnect,
+            hoconsc:mk(
+                emqx_schema:duration_ms(),
+                #{
+                    default => <<"2s">>,
+                    required => true,
+                    desc => ?DESC("auto_reconnect")
+                }
+            )},
+        %% Things related to sending messages to RabbitMQ
+        {exchange,
+            hoconsc:mk(
+                typerefl:binary(),
+                #{
+                    default => <<"messages">>,
+                    required => true,
+                    desc => ?DESC("exchange")
+                }
+            )},
+        {routing_key,
+            hoconsc:mk(
+                typerefl:binary(),
+                #{
+                    default => <<"my_routing_key">>,
+                    required => true,
+                    desc => ?DESC("routing_key")
+                }
+            )},
+        {delivery_mode,
+            hoconsc:mk(
+                typerefl:range(1, 2),
```

Comment:
Perhaps this could be:

```suggestion
                hoconsc:enum([persistent, non_persistent]),
```

and then transformed to 0,1 inside the connector?